webapp/src/components/open-api-plugins/plugin-wizard/PluginWizard.tsx

```diff
@@ -0,0 +1,227 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogBody,
+    DialogContent,
+    DialogSurface,
+    DialogTitle,
+    DialogTrigger,
+    Persona,
+    Text,
+    makeStyles,
+    tokens,
+} from '@fluentui/react-components';
+import { CheckmarkCircle48Regular, Dismiss24Regular } from '@fluentui/react-icons';
+import React, { ReactElement, useCallback, useState } from 'react';
+import AddPluginIcon from '../../../assets/plugin-icons/add-plugin.png';
+import { PluginManifest } from '../../../libs/models/PluginManifest';
+import { EnterManifestStep } from './steps/EnterManifestStep';
+import { usePlugins } from '../../../libs/hooks';
+import { ValidateManifestStep } from './steps/ValidateManifestStep';
+import { useDialogClasses } from '../../../styles';
+
+export const useClasses = makeStyles({
+    root: {
+        height: '400px',
+    },
+    surface: {
+        width: '500px',
+    },
+    center: {
+        paddingTop: '75px',
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        rowGap: tokens.spacingVerticalL,
+        'place-self': 'center',
+        width: '90%',
+    },
+});
+
+interface IWizardStep {
+    id: CreatePluginSteps;
+    header?: ReactElement;
+    body: ReactElement;
+    buttons?: ReactElement;
+}
+
+enum CreatePluginSteps {
+    EnterManifest,
+    ValidateManifest,
+    Confirmation,
+}
+
+export const PluginWizard: React.FC = () => {
+    const classes = useClasses();
+    const dialogClasses = useDialogClasses();
+    const plugins = usePlugins();
+
+    const [activeStep, setActiveStep] = useState(CreatePluginSteps.EnterManifest);
+    const [manifestDomain, setManifestDomain] = useState<string | undefined>();
+    const [manifestDomainError, setManifestDomainError] = useState<string | undefined>();
+    const [pluginValidated, setPluginValidated] = useState(false);
+    const [pluginManifest, setPluginManifest] = useState<PluginManifest | undefined>();
+
+    // Resets wizard to first step and blank slate
+    const resetLocalState = useCallback(() => {
+        setManifestDomain(undefined);
+        setManifestDomainError(undefined);
+        setActiveStep(CreatePluginSteps.EnterManifest);
+        setPluginValidated(false);
+    }, []);
+
+    const onAddPlugin = useCallback(() => {
+        if (pluginManifest && manifestDomain) {
+            plugins.addCustomPlugin(pluginManifest, manifestDomain);
+            setActiveStep(CreatePluginSteps.Confirmation);
+        } else {
+            setPluginValidated(false);
+            // TODO: [Issue #1973] add error handling
```

Comment:
should we update this to an issue in the new repo?

Reply:
I don't think it really matters right now since both are active and we're still tracking things on the old board